nvi-commons/src/main/java/no/sikt/nva/nvi/common/service/NviService.java

```diff
@@ -45,134 +35,122 @@ public static NviService defaultNviService() {
         return new NviService(defaultDynamoClient());
     }
 
-    //TODO: Remove JacocoGenerated when other if/else cases are implemented
-    @JacocoGenerated
-    public Optional<CandidateWithIdentifier> upsertCandidate(CandidateEvaluatedMessage evaluatedCandidate) {
-        if (!isExistingCandidate(evaluatedCandidate) && isNviCandidate(evaluatedCandidate)) {
-            return Optional.of(createCandidate(evaluatedCandidate));
-        } else if (isExistingCandidate(evaluatedCandidate) && isNviCandidate(evaluatedCandidate)) {
-            var existing = findByPublicationId(evaluatedCandidate.candidateDetails().publicationId()).orElseThrow();
+    @JacocoGenerated //TODO Temporary for coverage
+    public Optional<Candidate> upsertCandidate(DbCandidate candidate) {
```

Comment:
I would extract methods for readability:
```
    public Optional<Candidate> upsertCandidate(DbCandidate candidate) {
        if (isNotExistingCandidate(candidate.publicationId()) && isCandidate(candidate)) {
            return createPendingCandidate(candidate);
        } else if (isExistingCandidate(candidate.publicationId()) && isCandidate(candidate)) {
            var existing = findByPublicationId(candidate.publicationId()).orElseThrow();
            //TODO: Reset NVI Candidates here. See https://unit.atlassian.net/browse/NP-45113;
            return Optional.of(updateCandidate(existing.identifier(),
                                               candidate,
                                               generatePendingApprovalStatuses(
                                                   candidate.points())));
        }
        return Optional.empty();
    }

    private Optional<Candidate> createPendingCandidate(DbCandidate candidate) {
        return Optional.of(createCandidate(candidate, generatePendingApprovalStatuses(candidate.points())));
    }

    private static boolean isCandidate(DbCandidate candidate) {
        return candidate.applicable();
    }

    private boolean isNotExistingCandidate(URI publicationId) {
        return !isExistingCandidate(publicationId);
    }

    private boolean isExistingCandidate(URI publicationId) {
        return nviCandidateRepository.findByPublicationId(publicationId).isPresent();
    }
```

Reply:
Done